internal/datacoord/task_scheduler.go

```diff
@@ -297,3 +317,53 @@ func (s *taskScheduler) process(taskID UniqueID) bool {
 	}
 	return true
 }
+
+func (s *taskScheduler) collectTaskMetrics() {
+	defer s.wg.Done()
+
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-s.ctx.Done():
+			log.Warn("task scheduler context done")
+			return
+		case <-ticker.C:
+			s.RLock()
+			taskIDs := make([]UniqueID, 0, len(s.tasks))
+			for tID := range s.tasks {
+				taskIDs = append(taskIDs, tID)
+			}
+			s.RUnlock()
+
+			collectMetricsFunc := func(taskID int64) {
+				s.taskLock.Lock(taskID)
+				defer s.taskLock.Unlock(taskID)
+
+				task := s.getTask(taskID)
+				if task == nil {
+					s.taskLock.Unlock(taskID)
+					return
+				}
+
+				state := task.GetState()
+				switch state {
+				case indexpb.JobState_JobStateNone:
+					return
+				case indexpb.JobState_JobStateInit:
+					queueTime := time.Since(task.GetQueueTime())
```

Comment:
Please log warn if the pending or executing time of a task with a given taskID exceeds 6 hours ?

Reply:
I think 6 hours is too long, as it may delay the timely detection of issues. However, I made this duration configurable so it can be adjusted as needed. default duration is 5 minutes.